pkg/reconciler/pipelinerun/resources/validate_params.go

```diff
@@ -24,17 +24,18 @@ import (
 
 // ValidateParamTypesMatching validate that parameters in PipelineRun override corresponding parameters in Pipeline of the same type.
 func ValidateParamTypesMatching(p *v1beta1.PipelineSpec, pr *v1beta1.PipelineRun) error {
-	// Build a map of parameter names/types declared in p.
-	paramTypes := make(map[string]v1beta1.ParamType)
-	for _, param := range p.Params {
-		paramTypes[param.Name] = param.Type
+	// Build a map of parameter names/types declared in pr.
+	providedParams := make(map[string]v1beta1.ParamType)
+	for _, param := range pr.Spec.Params {
+		providedParams[param.Name] = param.Value.Type
 	}
 
-	// Build a list of parameter names from pr that have mismatching types with the map created above.
-	var wrongTypeParamNames []string
-	for _, param := range pr.Spec.Params {
-		if paramType, ok := paramTypes[param.Name]; ok {
-			if param.Value.Type != paramType {
+	// Build a list of parameter names from p that have mismatching types from corresponding types from p.
+	var wrongTypeParamNames []string 
+	for _, param := range p.Params {
+		paramType, ok := providedParams[param.Name]; 
+		if ok {
```

Comment:
`if paramType, ok := paramTypes[param.Name]; ok` is the syntax we usually prefer whenever we can get away with it